zk-token-sdk/src/sigma_proofs/mod.rs

```diff
@@ -15,15 +15,37 @@ pub mod pubkey_proof;
 pub mod zero_balance_proof;
 
 #[cfg(not(target_os = "solana"))]
-use {crate::errors::ProofVerificationError, curve25519_dalek::scalar::Scalar};
+use {
+    crate::errors::ProofVerificationError,
+    curve25519_dalek::{ristretto::CompressedRistretto, scalar::Scalar},
+};
 
+/// Deserializes an optional slice of bytes to a compressed Ristretto point.
+///
+/// This is a helper function for deserializing byte encodings of sigma proofs. It is designed to
+/// be used with `std::slice::Chunks`.
 #[cfg(not(target_os = "solana"))]
-fn canonical_scalar_from_slice(bytes: &[u8]) -> Result<Scalar, ProofVerificationError> {
-    if bytes.len() != 32 {
-        return Err(ProofVerificationError::Deserialization);
-    }
+fn ristretto_point_from_optional_slice(
+    optional_slice: Option<&[u8]>,
+) -> Result<CompressedRistretto, ProofVerificationError> {
+    let slice = optional_slice.ok_or(ProofVerificationError::Deserialization)?;
+    let point_bytes = slice[..32]
+        .try_into()
+        .map_err(|_| ProofVerificationError::Deserialization)?;
```

Comment:
```suggestion
    optional_slice
        .and_then(|slice| (slice.len() == RISTRETTO_POINT_LEN).then(|| slice)) // if chunk is the wrong length, convert to None
        .map(CompressedRistretto::from_slice)
        .ok_or(ProofVerificationError::Deserialization)
```
Since we're going to the trouble to avoid unwraps, we should check the length nicely, instead of potentially panicking on an invalid index.
Also, this one doesn't need the `try_into` clause, as `CompressedRistretto::from_slice` takes a slice.

Reply:
Much cleaner. Thanks!